### PR TITLE
bugfix: enable GNU/Linux support 🐧

### LINKED ISSUE #4: Linux support
This formular is available on [linuxbrew](http://linuxbrew.sh/) but it seems that the url always points to the Mac build. 🐧 

Besides from that: thank you very much for the homebrew package 🍺

### PR DESCRIPTION
**Pull request by alexanderadam**
Hi folks,

this PR finally adds GNU/Linux support to the homebrew package.
Thus it should fix #4.

Thank you for your work! :pray:

### CODE REVIEW

File: puma-dev.rb [hunk 1]
```diff
@@ -18,13 +26,29 @@ def install
   end
 
   def caveats
-    <<~EOS
+    caveats_info = <<~EOS
       Setup dev domains:
         sudo puma-dev -setup
```
Comment (nonrational) [inline puma-dev.rb:32]: `puma-dev -setup` is only available in the macOS build. 😞 I think we just want to refer people directly to https://github.com/puma/puma-dev#linux-support, as you've done below. 

I think you can nix the `caveats_info` var and just return one of two diff heredocs.

File: puma-dev.rb [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
   version '0.15.2'
 
+  base_url = "https://github.com/puma/puma-dev/releases/download/v#{version}/puma-dev-#{version}"
+
+  if OS.mac?
+    url "#{base_url}-darwin-amd64.zip"
+    sha256 '9b8efd851283f1ddb284580ac200c233ab9b285d770752d51d6d338f7c2020ab'
+  elsif OS.linux?
+    url "#{base_url}-linux-amd64.tar.gz"
+    sha256 '89995e01e07f30d5e41b2d9060c837a521228b0009a0ab037ee7ac066cc2403c'
```
Comment (nonrational) [inline puma-dev.rb:13]: 👍 Confirmed binary installation via a linuxbrew docker image.

**Review by nonrational — CHANGES_REQUESTED** [review]
Way cool! Also, I'm pretty sure this counts as an enhancement 😀

File: puma-dev.rb (hunk 1 above)
Comment (alexanderadam) [reply puma-dev.rb:32]: Good point. :+1: I just updated the PR.

**Comment by alexanderadam** [comment]
> Way cool! Also, I'm pretty sure this counts as an enhancement grinning

Thank you. Although I should have opened this PR instead of [the issue in 2017](https://github.com/puma/homebrew-puma/issues/4#issue-254666672). :wink: 
But I guess better late than never. :laughing: 

> `puma-dev -setup` is only available in the macOS build. :disappointed: I think we just want to refer people directly to https://github.com/puma/puma-dev#linux-support, as you've done below.

Please write me in case you plan to add `puma-dev -setup` support for GNU/Linux, too and I can help (without having to write Go).

**Review by nonrational — APPROVED** [review]

**Pull request merged**